### PR TITLE
Panel data bridge

### PR DESCRIPTION
**Pull request by ramyanaga**
Update panel to push-based data loading.
Potential Issue: not sure if store is loading all of the data correctly

### CODE REVIEW

File: app/components/Panel.js [hunk 1]
```diff
@@ -38,57 +39,49 @@ export default class Panel extends Component {
...
-          this.lastActionId = newLastActionId;
-          this.setState({
-            actionLog: result,
-          });
-        }
-      });
-    }, 100);
-
+     this.interval = setInterval(() => {
+     }, 100);
```
Comment (rrdelaney) [inline app/components/Panel.js]: Did you mean to remove this?

File: app/components/contentScript.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+chrome.runtime.sendMessage({
```
Comment (rrdelaney) [inline app/components/contentScript.js]: Was this file intended to be included?

File: extension/hook.js [hunk 3]
```diff
@@ -1,30 +1,58 @@
...
+      chrome.runtime.sendMessage({ APOLLO_CONNECTED: true}, function() {
+        passedApolloConnected = true;
+      });
+    }
+  }
+
+  if (!!event.data.trimmedObj) {
+    chrome.runtime.sendMessage({ trimmedObj: event.data.trimmedObj });
+  }
+  else {
```
Comment (rrdelaney) [inline extension/hook.js]: I don't think you need this `else` statement - a blank return should have the same effect as no return

File: app/components/Panel.js [hunk 4]
```diff
@@ -38,57 +39,49 @@ export default class Panel extends Component {
       automaticallyRunQuery: undefined
     };
 
+    let backgroundPageConnection = chrome.runtime.connect({
+      // sending tabId as name to make connection one-step process
+      name: chrome.devtools.inspectedWindow.tabId.toString()
+    });
+
+    backgroundPageConnection.onMessage.addListener((request, sender) => {
+      request = [request];
```
Comment (rrdelaney) [inline app/components/Panel.js]: Why wrap this in an array and then `.map` over it?

**Comment by stubailo** [comment]
@ramyanaga why is this adding 3000 lines of code? sounds like we included a file we don't need or something

**Comment by stubailo** [comment]
Ohh - package-lock.json from NPM 5. carry on.

File: app/components/Panel.js [hunk 5]
```diff
@@ -124,8 +117,6 @@ export default class Panel extends Component {
             state: window.__APOLLO_CLIENT__.queryManager.getApolloState(),
             dataWithOptimisticResults: window.__APOLLO_CLIENT__.queryManager.getDataWithOptimisticResults(),
           });
-
-          window.__APOLLO_CLIENT__.__actionHookForDevTools(logger);
```
Comment (stubailo) [inline app/components/Panel.js]: I think removing this might make the store panel stop working?

File: extension/background.js [hunk 6]
```diff
@@ -1,6 +1,44 @@
...
+let backgroundPageConnection = undefined;
+let requestLog = {};
+
+chrome.runtime.onConnect.addListener((port) => {
+  tabId = port.name
+  connections[tabId] = port;
+
+  port.onDisconnect.addListener(port => {
+
+    port.onMessage.removeListener(extensionListener);
```
Comment (stubailo) [inline extension/background.js]: I don't think extensionListener exists here.

File: extension/background.js [hunk 7]
```diff
@@ -1,6 +1,44 @@
...
+
+    const tabs = Object.keys(connections);
+    for (let i = 0; i < tabs.length; i++) {
+      if (connections[tabs[i]] == port) {
+        delete connections[tabs[i]];
```
Comment (stubailo) [inline extension/background.js]: I think this entire for loop thingy can just be replaced with

```js
delete connections[tabId];
```

Or am I missing something?

File: extension/background.js [hunk 8]
```diff
@@ -1,6 +1,44 @@
...
+  // not sure if being used
+  if (!apolloConnected && request.APOLLO_CONNECTED) {
     chrome.pageAction.show(sender.tab.id);
+    apolloConnected = true;
+  }
+  if (request.trimmedObj) {
+    try {
+      connections[sender.tab.id].postMessage(request.trimmedObj);
+    }
+    catch(err) {
```
Comment (stubailo) [inline extension/background.js]: When does this error occur? Can we instead add an `if` statement to detect that case? This will catch _all_ errors, not just the one we're specifically trying to handle here.

File: app/components/Panel.js (hunk 4 above)
Comment (ramyanaga) [inline app/components/Panel.js]: I tried to keep the code as similar to the old version. The comparable section of code in the old version returned the request through an eval in an array and mapped over it, so I ended up doing a similar thing.

Comment (stubailo) [inline app/components/Panel.js]: I think we should make this not an array.
